apps/nestjs-backend/src/features/record/record.service.ts

```diff
@@ -1347,9 +1349,150 @@ export class RecordService {
       .$queryRawUnsafe<{ __id: string }[]>(queryBuilder.toQuery());
     const ids = result.map((r) => r.__id);
 
+    // this search step should not abort the query
+    try {
+      const searchHitIndex = await this.getSearchHitIndex(queryBuilder, tableId, query);
+      return { ids, extra: { groupPoints, searchHitIndex } };
+    } catch (e) {
+      this.logger.error('Get search index error:', e);
```

Comment:
this.logger.error('Get search index error:' + (err as Error).message, (err as Error)?.stack);